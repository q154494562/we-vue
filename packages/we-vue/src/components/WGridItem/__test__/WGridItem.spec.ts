import { shallowMount } from '@vue/test-utils'
import GridItem from '../'

describe('grid-item', () => {
  test('create', () => {
    const wrapper = shallowMount(GridItem, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-grid-item')
    expect(wrapper.classes()).toContain('weui-grid')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('handle click', () => {
    const routeLinkSpy = jest.fn()
    const wrapper = shallowMount(GridItem, {
      propsData: {},
      methods: {
        routeLink: routeLinkSpy,
      },
    })

    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
    expect(routeLinkSpy).toHaveBeenCalled()
  })
})
