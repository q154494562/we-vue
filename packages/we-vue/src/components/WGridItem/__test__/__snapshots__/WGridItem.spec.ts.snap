// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid-item create 1`] = `<div class="weui-grid"></div>`;
